### DISCUSSION TITLE
Problema con los codigos de Colorama

### DISCUSSION THREAD

**Discussion by Juncruz9** [community/community#132832] [Programming Help]
### Select Topic Area

Question

### Body

........Tengo un problema en if modulo == colorama con los colores por que me los toma como Codigos de Escape ANSI, solo quiero que me solucionen los que tienen un solo corchete y no los que tienen dos cochetes, el f-string no me lo toma y no se que hacer
Gracias

import time
import sys
import random
import colorama
from colorama import Fore, Back, Style, deinit, init

# Inicializa colorama
init(autoreset=True)

def imprimir_con_efecto(texto, velocidad=0.05):
    """
    Imprime el texto carácter por carácter, simulando una escritura progresiva.

    :param texto: El texto que quieres imprimir.
    :param velocidad: Tiempo en segundos entre cada carácter (por defecto 0.05).
    """
    for char in texto:
        sys.stdout.write(char)
        sys.stdout.flush()  # Asegura que el texto se imprima inmediatamente
        time.sleep(velocidad)
    print()  # Imprime una nueva línea al final

def main():
    # Mensaje de bienvenida con efecto de escritura
    imprimir_con_efecto("Bienvenido al tutorial de módulos de Python.\n")
    
    # Solicita al usuario que elija un módulo para aprender más sobre él
    Modulo = input("Escriba el módulo que quiera aprender: ").strip().lower()
    
    if Modulo == "colorama":
        texto_colorama = f'''Facilita el uso de colores y estilos de texto en la consola
\nHacer que inicialice colorama 
  import colorama (importamos)
  from colorama import Fore, Back, Style, Cursor, deinit, init (agregamos todas las funciones que utilizaremos en colorama)
Utilizaremos init(autoreset=True) para que el programa funcione correctamente
\nCambiamos el color del texto con la función: (Fore.)
-Después del punto se pone un color: {Fore.BLACK}BLACK{Style.RESET_ALL}, {Fore.BLUE}BLUE{Style.RESET_ALL}, {Fore.CYAN}CYAN{Style.RESET_ALL}, {Fore.GREEN}GREEN{Style.RESET_ALL}, {Fore.LIGHTBLACK_EX}LIGHTBLACK_EX{Style.RESET_ALL}, {Fore.LIGHTBLUE_EX}LIGHTBLUE_EX{Style.RESET_ALL}, {Fore.LIGHTCYAN_EX}LIGHTCYAN_EX{Style.RESET_ALL}, {Fore.LIGHTGREEN_EX}LIGHTGREEN_EX{Style.RESET_ALL}, {Fore.LIGHTMAGENTA_EX}LIGHTMAGENTA_EX{Style.RESET_ALL}, {Fore.LIGHTRED_EX}LIGHTRED_EX{Style.RESET_ALL},
{Fore.LIGHTWHITE_EX}LIGHTWHITE_EX{Style.RESET_ALL}, {Fore.LIGHTYELLOW_EX}LIGHTYELLOW_EX{Style.RESET_ALL}, {Fore.MAGENTA}MAGENTA{Style.RESET_ALL}, {Fore.RED}RED{Style.RESET_ALL}, {Fore.RESET}RESET{Style.RESET_ALL}, {Fore.WHITE}WHITE{Style.RESET_ALL}, {Fore.YELLOW}YELLOW{Style.RESET_ALL}
ej--print(Fore.CYAN + "Este texto es de color CYAN" + Style.RESET_ALL)
    Console: {Fore.CYAN}Este texto es de color CYAN{Style.RESET_ALL}  
ej--print(f" {{Fore.GREEN}} GENERA {{Style.RESET_ALL}} una {{Fore.YELLOW}} PALABRA {{Style.RESET_ALL}} o varias palabras ({{Fore.BLUE}} JUNTAS O SEPARADAS {{Style.RESET_ALL}}),
    en este caso elegí el color {{Fore.RED}} ROJO {{Style.RESET_ALL}}")
    Console: {Fore.GREEN} GENERA {Style.RESET_ALL} una {Fore.YELLOW} PALABRA {Style.RESET_ALL} o varias palabras ({Fore.BLUE} JUNTAS O SEPARADAS {Style.RESET_ALL}), en este caso elegí el color {Fore.RED} ROJO {Style.RESET_ALL}
 \nCambiamos el fondo del texto con la función: (Back.)
-Después del punto se pone un color: {Fore.BLACK}BLACK{Style.RESET_ALL}, {Fore.BLUE}BLUE{Style.RESET_ALL}, {Fore.CYAN}CYAN{Style.RESET_ALL}, {Fore.GREEN}GREEN{Style.RESET_ALL}, {Fore.LIGHTBLACK_EX}LIGHTBLACK_EX{Style.RESET_ALL}, {Fore.LIGHTBLUE_EX}LIGHTBLUE_EX{Style.RESET_ALL}, {Fore.LIGHTCYAN_EX}LIGHTCYAN_EX{Style.RESET_ALL}, {Fore.LIGHTGREEN_EX}LIGHTGREEN_EX{Style.RESET_ALL}, {Fore.LIGHTMAGENTA_EX}LIGHTMAGENTA_EX{Style.RESET_ALL}, {Fore.LIGHTRED_EX}LIGHTRED_EX{Style.RESET_ALL},
{Fore.LIGHTWHITE_EX}LIGHTWHITE_EX{Style.RESET_ALL}, {Fore.LIGHTYELLOW_EX}LIGHTYELLOW_EX{Style.RESET_ALL}, {Fore.MAGENTA}MAGENTA{Style.RESET_ALL}, {Fore.RED}RED{Style.RESET_ALL}, {Fore.RESET}RESET{Style.RESET_ALL}, {Fore.WHITE}WHITE{Style.RESET_ALL}, {Fore.YELLOW}YELLOW{Style.RESET_ALL}
ej--print(Back.BLUE + "Este texto tiene un fondo de color BLUE" + Style.RESET_ALL)
    Console: {Back.BLUE}Este texto tiene un fondo de color BLUE{Style.RESET_ALL}
ej--print(f" {{Back.RED}} GENERA {{Style.RESET_ALL}} una {{Back.LIGHTBLUE_EX}} PALABRA {{Style.RESET_ALL}} o varias palabras ({{Back.MAGENTA}} JUNTAS O SEPARADAS {{Style.RESET_ALL}}),
    en este caso elegí el color {{Fore.GREEN}} VERDE {{Style.RESET_ALL}}")
    Console: {Back.RED} GENERA {Style.RESET_ALL} una {Back.LIGHTBLUE_EX} PALABRA {Style.RESET_ALL} o varias palabras ({Back.MAGENTA} JUNTAS O SEPARADAS {Style.RESET_ALL}), en este caso elegí el color {Back.GREEN} VERDE {Style.RESET_ALL}
\nCerrar cada función del módulo y hacer más brillantes los textos
{{Style.RESET_ALL}} ..Asegura que los estilos no vayan más allá del punto deseado en la salida, cerrando cada función
{{Style.BRIGHT}} ..Hace que los textos sean más brillantes y no se suele notar tanto, es recomendable acompañarlo con algún color!
'''
        imprimir_con_efecto(texto_colorama, velocidad=0.05)
    
    elif Modulo == "random":
        texto_random = f'''Proporciona herramientas para generar números aleatorios y seleccionar elementos aleatoriamente
\nGenera un número aleatorio entre el 0 y el 1: (.random)
numero_aleatorio = random.random()
print(numero_aleatorio)
Console: 0.08758774866699959
\nGenera un número aleatorio entre un número y otro: (.randint)
numero_aleatorio = random.randint(1, 10)
print(numero_aleatorio)
Console: 8
\nGenera un número flotante (con coma) aleatorio entre un número y otro: (.uniform)
numero_aleatorio = random.uniform(5, 10)
print(numero_aleatorio)
Console: 7.643994879688099
\nObtiene la suma de un mismo número específico hasta un límite específico aleatoriamente: (.randrange)
print(random.randrange(10, 140, 20))
print(random.randrange("a,b,c"))
a = "Se le suma al número original c"
b = "se restará b - c para ver el límite de aleatoriedad y luego se le sumará c + a, siendo el límite 130" 
c = "Siempre estará habilitado el 0 y se utilizará toda la tabla del número c (20,40,60,etc), siendo el límite b - c"
(10, 140, 20)
1_ Puede salir aleatoriamente 0, 20, 40, 60, 80, 100 o 120 (140 no tocará nunca) b - c
2_ Ahora se le suma a (+10) c + a
3_ Puede salir aleatoriamente 0, 30, 50, 70, 90, 110 o 130
Console: 110
\nObtiene un elemento aleatorio de una lista o tupla: (.choice)
colores = ["rojo","verde","azul","amarillo"]
color_elegido = random.choice(colores)
print("Color elegido aleatoriamente:", color_elegido)
Console: Color elegido aleatoriamente: verde
\nObtiene varios elementos aleatorios de una lista o tupla: (.sample)
cartas = ["As","2","3","4","5","6","7","8","9","10","J","Q","K"]
mano = random.sample(cartas, 5)
print("Mano de cartas:", mano)
Console: Mano de cartas: ['10','2','As','5','Q']
\nSelecciona el número aleatorio generado: (.getstate, .setstate)
state = random.getstate()
numero_aleatorio = random.uniform(5, 10)
print("Número aleatorio generado:", numero_aleatorio)
random.setstate(state)
mismo_numero_aleatorio = random.uniform(5, 10)
print("El mismo número aleatorio:", mismo_numero_aleatorio)
Console: Número aleatorio generado: 7.004452552911907
         El mismo número aleatorio: 7.004452552911907
\nBorra el número generado aleatoriamente y lo cambia por otro:
numero_aleatorio = random.uniform(5, 10)
print("Número aleatorio generado:", numero_aleatorio)
nuevo_numero_aleatorio = random.uniform(5, 10)
print("Nuevo número aleatorio:", nuevo_numero_aleatorio)
'''
        imprimir_con_efecto(texto_random, velocidad=0.05)
    
    else:
        imprimir_con_efecto("Módulo no reconocido. Por favor, escriba de nuevo.\n", velocidad=0.05)
        
    # Mensaje de cierre con efecto de escritura
    imprimir_con_efecto("Por favor reinicia y extiende la consola para una mejor experiencia.\n", velocidad=0.05)
    
    # Desactiva colorama
    deinit()

# Llama a la función principal
main()

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Juncruz9 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
